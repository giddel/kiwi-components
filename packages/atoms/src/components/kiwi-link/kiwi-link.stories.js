import { withKnobs, text } from '@storybook/addon-knobs';
import { html } from 'lit-html';
import notes from './readme.md';

export default {
  title: 'KiwiComponents/Link',
  parameters: { notes },
  decorators: [withKnobs],
};

export const basic = () => {
  const to = text('to', 'https://www.kiwigrid.com');
  const label = text('label', 'Click Me!');

  return html` <div class="panel panel-default">
    <div class="panel-body">
      <kiwi-link to="${to}">${label}</kiwi-link>
    </div>
  </div>`;
};
