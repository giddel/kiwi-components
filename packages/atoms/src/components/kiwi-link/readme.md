# kiwi-link



<!-- Auto Generated Below -->


## Properties

| Property          | Attribute | Description                                                            | Type                        | Default     |
| ----------------- | --------- | ---------------------------------------------------------------------- | --------------------------- | ----------- |
| `to` _(required)_ | `to`      | The navigation destination. Can be either a url or a `NavigateToEvent` | `NavigateToEvent \| string` | `undefined` |


## Events

| Event        | Description                                | Type                           |
| ------------ | ------------------------------------------ | ------------------------------ |
| `navigateTo` | Dummy event documentation - please fix me! | `CustomEvent<NavigateToEvent>` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
