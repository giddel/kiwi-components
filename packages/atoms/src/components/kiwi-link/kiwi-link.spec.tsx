import { newSpecPage } from '@stencil/core/testing';
import { KiwiLink } from './kiwi-link';

describe('kiwi-link', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [KiwiLink],
      html: `<kiwi-link to="https://google.com">Google</kiwi-link>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
