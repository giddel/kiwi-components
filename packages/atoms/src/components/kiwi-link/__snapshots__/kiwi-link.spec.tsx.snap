// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kiwi-link renders 1`] = `
<kiwi-link to="https://google.com">
  <!---->
  <a href="https://google.com">
    Google
  </a>
</kiwi-link>
`;
